### ISSUE TITLE
flutter run after flutter create does not work on ios simulator

### ISSUE THREAD

**Issue by drewwarren** [flutter/flutter#5727]
## Steps to Reproduce

This occurs on both master and alpha, although with different symptoms between the two. I am mainly concerned about master as there is an existing bug that I think references the alpha issues (see below).

### Master

```
~$ rm -rf flutter_test_async/
~$ mkdir flutter_test_async
~$ cd flutter_test_async/
flutter_test_async$ flutter create .
... output from flutter create
flutter_test_async$ flutter run -v
... lots of other output from run
          sent 29 bytes  received 20 bytes  98.00 bytes/sec
          total size is 0  speedup is 0.00
          
          Touch build/Release-iphonesimulator/Runner.app
              cd /Users/drewwarren/flutter_test_async/ios
              export PATH="/Applications/Xcode.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/usr/bin:/Applications/Xcode.app/Contents/Developer/usr/bin:/Users/drewwarren/bin:/Users/drewwarren/flutter/bin/cache/dart-sdk/bin:/Users/drewwarren/flutter/bin:/usr/local/git/current/bin:/usr/local/bin:/usr/bin:/bin:/usr/local/sbin:/usr/sbin:/sbin"
              /usr/bin/touch -c /Users/drewwarren/flutter_test_async/ios/build/Release-iphonesimulator/Runner.app
          
          ** BUILD SUCCEEDED **
   0 ms • Could not find the built application bundle at build/ios/Release-iphonesimulator/Runner.app.
   1 ms • Error running application on iPhone 5s.
   0 ms • 'flutter run' took 3435ms; exiting with code 2.
   0 ms • ensureAnalyticsSent: 52ms
```

```
[✓] Flutter (on Mac OS, channel master)
    • Flutter at /Users/drewwarren/flutter
    • Framework revision f6f37ef260 (3 hours ago), 2016-09-02 10:46:46
    • Engine revision dcb026188a
    • Tools Dart version 1.19.0-dev.5.0

[x] Android toolchain - develop for Android devices
    x Android Studio / Android SDK not found. Download from https://developer.android.com/sdk/
      (or visit https://flutter.io/setup/#android-setup for detailed instructions).

[✓] iOS toolchain - develop for iOS devices (Xcode 7.3.1)
    • XCode at /Applications/Xcode.app/Contents/Developer
    • Xcode 7.3.1, Build version 7D1014

[✓] Atom - a lightweight development environment for Flutter
    • flutter plugin version 0.2.4
    • dartlang plugin version 0.6.37

[✓] Connected devices
    • iPhone 5s • 16BCCC3C-2F9C-401F-B2BF-97F1E3D2929A • ios
```

### Alpha

The tail of the run -v output on alpha just repeats the same stuff. On alpha this is likely the same issue as #5723.

```
 20 ms • Unable to locate an Android SDK.
   6 ms • xcode-select --print-path
   0 ms • Exit code 0 from: xcode-select --print-path
   0 ms • /Applications/Xcode.app/Contents/Developer
  56 ms • xcodebuild -version
   0 ms • Exit code 0 from: xcodebuild -version
   0 ms • Xcode 7.3.1
          Build version 7D1014
  17 ms • xcrun clang
   2 ms • Unable to locate an Android SDK.
   7 ms • /usr/local/bin/adb devices -l
   0 ms • Exit code 0 from: /usr/local/bin/adb devices -l
   2 ms • List of devices attached
  10 ms • idevice_id -h
   3 ms • which idevice_id
   0 ms • Exit code 0 from: which idevice_id
   0 ms • /usr/local/bin/idevice_id
   4 ms • /usr/local/bin/idevice_id -l
   1 ms • Exit code 0 from: /usr/local/bin/idevice_id -l
  85 ms • /usr/bin/xcrun simctl list --json devices
  32 ms • /usr/bin/defaults read /Users/drewwarren/flutter_test_async/ios/Runner/Info CFBundleIdentifier
   0 ms • Exit code 0 from: /usr/bin/defaults read /Users/drewwarren/flutter_test_async/ios/Runner/Info CFBundleIdentifier
   1 ms • com.yourcompany.flutterTestAsync
  15 ms • Stopping app 'Runner.app' on iPhone 5s.
   2 ms • Running lib/main.dart on iPhone 5s...
   0 ms • Building Runner.app for 16BCCC3C-2F9C-401F-B2BF-97F1E3D2929A.
   2 ms • tail -n 0 -F /Users/drewwarren/Library/Logs/CoreSimulator/16BCCC3C-2F9C-401F-B2BF-97F1E3D2929A/system.log
  38 ms • Building build/app.flx
 142 ms • tail -n 0 -F /private/var/log/system.log
   4 ms • which zip
  26 ms • Encoding zip file to build/app.flx
 187 ms • zip -q /Users/drewwarren/flutter_test_async/build/app.flx fonts/MaterialIcons-Regular.ttf fonts/Roboto-Regular.ttf fonts/Roboto-Italic.ttf fonts/Roboto-Thin.ttf…
   0 ms • Exit code 0 from: zip -q /Users/drewwarren/flutter_test_async/build/app.flx fonts/MaterialIcons-Regular.ttf fonts/Roboto-Regular.ttf fonts/Roboto-Italic.ttf fonts/Roboto-Thin.ttf fonts/Roboto-ThinItalic.ttf fonts/Roboto-Light.ttf fonts/Roboto-LightItalic.ttf fonts/Roboto-Medium.ttf fonts/Roboto-MediumItalic.ttf fonts/Roboto-Bold.ttf fonts/Roboto-BoldItalic.ttf fonts/Roboto-Black.ttf fonts/Roboto-BlackItalic.ttf fonts/RobotoCondensed-Regular.ttf fonts/RobotoCondensed-Italic.ttf fonts/RobotoCondensed-Light.ttf fonts/RobotoCondensed-LightItalic.ttf fonts/RobotoCondensed-Bold.ttf fonts/RobotoCondensed-BoldItalic.ttf AssetManifest.json FontManifest.json LICENSE
   0 ms • Built build/app.flx.
  97 ms • /usr/bin/xcrun simctl get_app_container booted com.yourcompany.flutterTestAsync
  11 ms • /usr/bin/killall Runner
   0 ms • Using engine artifacts dir: /Users/drewwarren/flutter/bin/cache/artifacts/engine/ios
  60 ms • xcodebuild -version
   0 ms • Exit code 0 from: xcodebuild -version
   1 ms • Xcode 7.3.1
          Build version 7D1014
   2 ms • Trying to resolve native pub services.
   1 ms • Looking for YAML at 'flutter.yaml'
   0 ms • No services specified in the manifest
   2 ms • Found 0 service definition(s).
   0 ms • Copying service frameworks to '/Users/drewwarren/flutter_test_async/ios/Frameworks'.
   2 ms • Creating service definitions manifest at 'ios/ServiceDefinitions.json'
   2 ms • /usr/bin/env xcrun xcodebuild clean build -configuration Release ONLY_ACTIVE_ARCH=YES -workspace Runner.xcworkspace -scheme Runner BUILD_DIR=/Users/drewwarren/flutter_test_async/ios/build -sdk iphonesimulator -arch x86_64
...
2103 ms • [SYS LOG] Sep  2 13:29:15 drewwarren-macbookpro com.apple.CoreSimulator.CoreSimulatorService[33216]: Requested phone is is not available for pairing.  Assuming it has the capability.
  25 ms • [SYS LOG] Sep  2 13:29:17 drewwarren-macbookpro com.apple.usbmuxd[77613]: HandleUSBMuxDictionary client 0x7f8d43402470-libusbmuxd/org.libimobiledevice.usbmuxd using library usbmuxd built for freedom, running usbmuxd-364
  38 ms • [SYS LOG] Sep  2 13:29:17 drewwarren-macbookpro com.apple.usbmuxd[77613]: HandleUSBMuxDictionary client 0x7f8d43400c50-libusbmuxd/org.libimobiledevice.usbmuxd using library usbmuxd built for freedom, running usbmuxd-364
   0 ms • [SYS LOG] Sep  2 13:29:17 drewwarren-macbookpro com.apple.CoreSimulator.CoreSimulatorService[33216]: Requested watch is is not available for pairing.  Assuming it has the capability.
   0 ms • [SYS LOG] Sep  2 13:29:17 drewwarren-macbookpro com.apple.CoreSimulator.CoreSimulatorService[33216]: Requested phone is is not available for pairing.  Assuming it has the capability.
   0 ms • [SYS LOG] Sep  2 13:29:17 drewwarren-macbookpro com.apple.CoreSimulator.CoreSimulatorService[33216]: Requested watch is is not available for pairing.  Assuming it has the capability.
  16 ms • [SYS LOG] Sep  2 13:29:17 drewwarren-macbookpro com.apple.CoreSimulator.CoreSimulatorService[33216]: Requested phone is is not available for pairing.  Assuming it has the capability.
   0 ms • [SYS LOG] Sep  2 13:29:17 drewwarren-macbookpro com.apple.CoreSimulator.CoreSimulatorService[33216]: Requested watch is is not available for pairing.  Assuming it has the capability.
   0 ms • [SYS LOG] Sep  2 13:29:17 drewwarren-macbookpro com.apple.CoreSimulator.CoreSimulatorService[33216]: Requested phone is is not available for pairing.  Assuming it has the capability.
   0 ms • [SYS LOG] Sep  2 13:29:17 drewwarren-macbookpro com.apple.CoreSimulator.CoreSimulatorService[33216]: Requested watch is is not available for pairing.  Assuming it has the capability.
... repeats indefinitely.
```

**Comment by drewwarren** [comment]
doing a bisect b5f763b is the commit that broke this (on master)

**Comment by eseidelGoogle** [comment]
Thanks for the reduction and bug @drewwarren.

Are either @cbracken or @danrubel available to fix this on master?

**Comment by cbracken** [comment]
I'll grab this.

**Comment by cbracken** [comment]
Fixed by #5732
